shadowsocks-libev/files/shadowsocks.track

```diff
@@ -0,0 +1,386 @@
+#! /usr/bin/env lua
+-- Copyright 2015 OVH <OverTheBox@ovh.net>
+-- Simon Lelievre <simon.lelievre@corp.ovh.com>
+-- Sebastien Duponcheel <sebastien.duponcheel@ovh.net>
+--
+-- This file is part of OverTheBox for OpenWrt.
+--
+--    OverTheBox is free software: you can redistribute it and/or modify
+--    it under the terms of the GNU General Public License as published by
+--    the Free Software Foundation, either version 3 of the License, or
+--    (at your option) any later version.
+--
+--    OverTheBox is distributed in the hope that it will be useful,
+--    but WITHOUT ANY WARRANTY; without even the implied warranty of
+--    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--    GNU General Public License for more details.
+--
+--    You should have received a copy of the GNU General Public License
+--    along with OverTheBox.  If not, see <http://www.gnu.org/licenses/>
+--
+--	Contributor : Jean Labrousse <jlabrous@github.com>
+--
+
+
+local p   = require 'posix'
+local sig = require "posix.signal"
+
+local sys	= require("luci.sys")
+local json      = require("luci.json")
+local math      = require("math")
+
+local method -- ping function bindings
+local fallback_method -- fall
+
+local function handle_exit()
+	p.closelog()
+	os.exit();
+end
+
+function tlog (tbl, indent)
+  if not indent then indent = 0 end
+  if not tbl then return end
+  for k, v in pairs(tbl) do
+    formatting = string.rep("  ", indent) .. k .. ": "
+    if type(v) == "table" then
+      log(formatting)
+      tlog(v, indent+1)
+    elseif type(v) == 'boolean' then
+      log(formatting .. tostring(v))
+    else
+      log(formatting .. v)
+    end
+  end
+end
+
+function socks_request( host, interface, timeout, port )
+	local fd, err = p.socket(p.AF_INET, p.SOCK_STREAM, 0)
+	if not fd then return fd, err end
+
+	p.bind (fd, { family = p.AF_INET, addr = "0.0.0.0", port = 0 })
+
+	-- timeout on socket
+	local ok, err = p.setsockopt(fd, p.SOL_SOCKET, p.SO_RCVTIMEO, math.floor(timeout/1000), (timeout % 1000) * 1000 )
+	if not ok then
+		if fd then p.close(fd) end
+		return ok, err
+	end
+
+	local ok, err = p.setsockopt(fd, p.SOL_SOCKET, p.SO_SNDTIMEO, math.floor(timeout/1000), (timeout % 1000) * 1000 )
+	if not ok then
+		if fd then p.close(fd) end
+		return ok, err
+	end
+
+	local r, err = p.getaddrinfo (host, port, { family = p.AF_INET, socktype = p.SOCK_STREAM })
+	if not r then
+		if fd then p.close(fd) end
+		return false, err
+	end
+
+	local t1 = p.clock_gettime(p.CLOCK_REALTIME)
+
+	local ok, err, e = p.connect (fd, r[1])
+	if not ok then
+		if fd then p.close(fd) end
+		return ok, err
+	end
+
+	if fd then
+		local data, sa, err
+		while (diff_nsec(t1, p.clock_gettime(p.CLOCK_REALTIME) )/1000) < timeout do
+			data, sa, err = p.recvfrom(fd, 1024)
+			if data then
```

Comment:
You shouldn't drop received packets this way, it's a stream so you should merge them and try again.
You may want to call curl (or libcurl) to do the job, this will be a safer solution.